src/ussd_nurse_rapidpro.js

```diff
@@ -370,7 +371,161 @@ go.app = function() {
+            return new FreeText(name, {
+                question: question,
+                check: function(content) {
+                    if (!utils.check_valid_number(content)
+                        || content.length !== 8) {
+                        return error;
+                    } else {
+                        /*
+                            add function to update info against operator id
+                            then lead to state where user details are successfully changed
+                        */
+                        self.im.user.set_answer("sanc_reg_no", content);
+                        return null;
+                    }
+                },
+                next: 'state_end_detail_changed',
+            });
+        });
 
+        self.states.add('state_end_detail_changed', function(name) {
+            return new EndState(name, {
+                text: $("Thank you. Your NurseConnect details have been changed. To change any other details, please dial {{channel}} again.")
+                    .context({channel: self.im.config.channel}),
+                next: 'state_start',
+             });
+        });
+
+        self.states.add('state_change_id_no', function(name) {
+            var question = $("Please select your type of identification:");
+            return new ChoiceState(name, {
+                question: question,
+                choices: [
+                    new Choice('state_id_no', $('RSA ID')),
+                    new Choice('state_passport', $('Passport'))
+                ],
+                next: function(choice) {
+                    return choice.value;
+                }
+            });
+        });
+
+        self.states.add('state_id_no', function(name) {
+            var error = $("Sorry, the format of the ID number is not correct. Please enter your RSA ID number again, e.g. 7602095060082");
+            var question = $("Please enter your 13-digit RSA ID number:");
+            return new FreeText(name, {
+                question: question,
+                check: function(content) {
+                    if (!utils.validate_id_za(content)) {
+                        return error;
+                    }
+                    else{
+                        /*
+                        add function to update id type info against operator id
+                        add function to update id number info against operator id
+                        add function to update date of birth info against operator id
+                        then lead to state where user details are successfully changed
+                        */
+                    var date_of_birth = utils.extract_za_id_dob(content);
+                    self.im.user.set_answer("id_type", "sa_id");
+                    self.im.user.set_answer("sa_id_no", content);
+                    self.im.user.set_answer("dob", date_of_birth);
```

Comment:
The `id_type` will be saved under `state_change_id_no`, and the rest under `state_id_no`, so we don't need to save them again.

Reply:
Yes, I think we can do the date of birth generation when we send the update to RapidPro, instead of here.

Reply:
noted